### ISSUE TITLE
Support STS AssumeRole `SourceIdentity` value

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/aws-sdk-go-base#87]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When assuming an IAM Role, a [`SourceIdentity`](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html#API_AssumeRole_RequestParameters) value can be specified.
Allow this value to be specified by this library's users.

### Potential Terraform Backend/Provider Configuration

```hcl
provider "aws" {
  assume_role {
    source_identity = "Alice"
  }
}
```

### References

* https://github.com/hashicorp/terraform-provider-aws/issues/22128

**Comment by gdavison** [comment]
This will need aws/aws-sdk-go-v2#575 to be resolved

**Comment by gdavison** [comment]
Implemented in aws/aws-sdk-go-v2#1588

**Comment by wlami** [comment]
Thanks a lot for merging this, @gdavison! 
I'd like to use this feature in https://github.com/hashicorp/terraform-provider-aws/issues/22128 , is there any timeline when new versions/tags of this module will be published?
